paasta_tools/tron_tools.py

```diff
@@ -282,6 +282,32 @@ def __init__(
         self.job, self.action = decompose_instance(instance)
         # Indicate whether this config object is created for validation
         self.for_validation = for_validation
+        self.action_spark_config = None
+
+    def get_cpus(self) -> float:
+        # set Spark driver pod CPU if it is specified by Spark arguments
+        cpus = 0.0
+        if (
+            self.action_spark_config
+            and "spark.driver.cores" in self.action_spark_config
+        ):
+            cpus = float(self.action_spark_config["spark.driver.cores"])
+        # use the soa config otherwise
+        return cpus or super().get_cpus()
```

Comment:
```suggestion
    def get_cpus(self) -> float:
        if (
            self.get_executor() == "spark"
            and "spark.driver.cores" in self.action_spark_config
        ):
            return self.action_spark_config["spark.driver.cores"]
        # NOTE: we fallback to this if there's no spark.driver.cores config to
        # use the paasta default
        return super().get_cpus()
```

Reply:
that said, maybe we want a different default CPU for spark drivers? the paasta default for both cpu and memory might be too small for drivers :)

Reply:
(also, imo we should add `spark.driver.cores` to the spark_args schema and enforce that it's a float there so that we don't need to remember to cast it in the python code :))

Reply:
I would rather handle enforce the default for `spark.driver.cores` and `spark.driver.memory` in service_configuration_lib than using another default value here.
Even if we enforce the float in schema, it passes through operations in service_configuration_lib and everything is returned as a string

Reply:
we need this to return the resources that are actually used tho (either user-specified or the defaults that are used) - this can call `service_configuration_lib` code if necessary :)

re: s_c_l: i think it's also worth slowly refactoring that code - it's sorta painful to work with when everything in that library is stringly-typed :)